src/components/App/App.js

```diff
@@ -157,18 +157,15 @@ const App = () => {
 
       const user = await getCurrentUser();
       updateUser(user);
-
+      const favMentorsFromLocalStorage = readAndUpdateFavMentorsFromLocalStorage();
       Promise.all([
         user &&
-        getFavorites().then(setFavorites),
+        getFavorites().then((favorites) => {
+          setFavorites([...favMentorsFromLocalStorage, ...favorites]);
```

Comment:
Consider to clean duplicates.
https://stackoverflow.com/a/14438954/863110

Reply:
Thanks for pointing this out. I have done the changes. The changes even prevent the updateMentor API call to the server for duplicate entries.